src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceService.kt

```diff
@@ -15,16 +19,20 @@ class LicenceService(
       throw NoDataFoundException("licence", "booking id", bookingId)
     }
 
-    val licenceData = licences.first().licence ?: throw NoDataFoundException("licence data", "booking id", bookingId)
+    val licence = licences.first()
+
+    val licenceData = licence.licence ?: throw NoDataFoundException("licence data", "booking id", bookingId)
 
     val cas2Referral = licenceData.bassReferral
     val curfew = licenceData.curfew
     val proposedAddress = licenceData.proposedAddress
 
     return HdcLicence(
+      licenceId = licence.id,
       curfewAddress = getAddress(curfew, cas2Referral, proposedAddress),
       firstNightCurfewHours = curfew?.firstNight,
-      curfewHours = curfew?.curfewHours,
+      // curfewHours referred to as curfewTimes in CVL as going forward a more suitable name and had to distinguish between the two different curfew data formats
+      curfewTimes = formatCurfewHoursObject(curfew?.curfewHours!!, licence.id),
```

Comment:
**suggestion:** This will now blow up if there are no curfew section whereas previously it would pass back null.

it's probably best to use the `?.let` syntax to only format the curfew times if curfewHours is not null. 

You could also use `?:` to return an empty list if it is null